src/components/application_manager/test/commands/mobile/send_location_request_test.cc

```diff
@@ -0,0 +1,412 @@
+    (*message_)[strings::params][strings::connection_key] = kConnectionKey;
+    (*message_)[strings::params][strings::function_id] = kFunctionID;
+    (*message_)[strings::msg_params] =
+        SmartObject(smart_objects::SmartType_Map);
+    (*message_)[strings::msg_params][strings::address] = kCorrectAddress;
+    EXPECT_CALL(app_mngr_, application(kConnectionKey))
+        .WillOnce(Return(mock_app_));
+
+    AllowMandatoryFields();
+  }
+
+  void HMICapabilitiesSetupHelper() {
+    EXPECT_CALL(app_mngr_, hmi_capabilities())
+        .WillOnce(ReturnRef(mock_hmi_capabilities_));
+    EXPECT_CALL(mock_hmi_capabilities_, is_ui_cooperating())
+        .WillOnce(Return(true));
+  }
+
+  void HMICapabilitiesSetupWithArguments(
+      Common_TextFieldName::eType field_name) {
+    HMICapabilitiesSetupHelper();
+    (*disp_cap_)[hmi_response::text_fields] =
+        SmartObject(smart_objects::SmartType_Array);
+    (*disp_cap_)[hmi_response::text_fields][0] =
+        SmartObject(smart_objects::SmartType_Map);
+    (*disp_cap_)[hmi_response::text_fields][0][strings::name] = field_name;
+    EXPECT_CALL(mock_hmi_capabilities_, display_capabilities())
+        .Times(2)
+        .WillRepeatedly(Return(disp_cap_.get()));
+  }
+
+  void FinishSetup() {
+    EXPECT_CALL(*mock_app_, hmi_app_id()).WillOnce(Return(kAppID));
+    EXPECT_CALL(app_mngr_,
+                ManageHMICommand(HMIResultCodeIs(
+                    hmi_apis::FunctionID::Navigation_SendLocation)));
+  }
+
+  void FinishSetupCancelled(mobile_apis::Result::eType result) {
+    EXPECT_CALL(*mock_app_, hmi_app_id()).Times(0);
+    EXPECT_CALL(app_mngr_, ManageHMICommand(_)).Times(0);
+    EXPECT_CALL(app_mngr_, ManageMobileCommand(MobileResultCodeIs(result), _));
+  }
+
+  void AllowMandatoryFields() {
+    application_manager::CommandParametersPermissions& permissions =
+        command_->get_parameters_permissions();
+    permissions.allowed_params.push_back(strings::longitude_degrees);
+    permissions.allowed_params.push_back(strings::latitude_degrees);
+  }
+
+  MockAppPtr mock_app_;
+  MockHMICapabilities mock_hmi_capabilities_;
+  MockMessageHelper& mock_message_helper_;
+  SharedPtr<SmartObject> disp_cap_;
+  MessageSharedPtr message_;
+  CommandSPrt command_;
+};
+
+TEST_F(SendLocationRequestTest, Run_InvalidApp_Success) {
+  (*message_)[strings::params][strings::connection_key] = kConnectionKey;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(MockAppPtr()));
+  FinishSetupCancelled(mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+  command_->Run();
+}
+
+TEST_F(SendLocationRequestTest, Run_DeliveryMode_Success) {
+  InitialSetup(message_);
+  SmartObject msg_params = (*message_)[strings::msg_params];
+  msg_params[strings::delivery_mode] = SmartObject();
+  (*message_)[strings::msg_params] = msg_params;
+  msg_params.erase(strings::delivery_mode);
+  EXPECT_CALL(mock_message_helper_, PrintSmartObject(_))
```

Comment:
@indlu, please ignore call of `PrintSmartObject()`. At first - it's redundant, and second - later-on it will be removed as in `pre_5.0.0` branch.